third_party/terraform/resources/resource_storage_bucket.go

```diff
@@ -1110,32 +1099,8 @@ func resourceGCSBucketLifecycleRuleConditionHash(v interface{}) int {
 		buf.WriteString(fmt.Sprintf("%s-", v.(string)))
 	}
 
-	// Note that we are keeping the boolean notation from when is_live was
-	// the only field (i.e. not deprecated) in order to prevent a diff from
-	// hash key.
-	// There are three possible states for the actual condition
-	// and correspond to the following hash codes:
-	//
-	// 1. LIVE only: "true-"
-	//    Applies for one of:
-	//      with_state = "" && is_live = true
-	//      with_state = "LIVE"
-	//
-	// 2. ARCHIVED only: "false-"
-	//    Applies for one of:
-	//      with_state = "" && is_live = false
-	//      with_state = "ARCHIVED"
-	//
-	// 3. ANY (i.e. LIVE and ARCHIVED): ""
-	//    Applies for one of:
-	//      with_state = "ANY"
-
 	withStateV, withStateOk := m["with_state"]
-	if !withStateOk || withStateV.(string) == "" {
-		if isLiveV, ok := m["is_live"]; ok {
-			buf.WriteString(fmt.Sprintf("%t-", isLiveV.(bool)))
-		}
-	} else if withStateOk {
+	if withStateOk {
```

Comment:
The goal here is to write nothing to the hash code if the state is "ANY"?

Reply:
That's how I interpreted the note that I removed (and maybe should have left in?).  But I could have misinterpreted it.

Reply:
I think it makes sense, just want to make sure there isn't a case not handled.

If `withStateV.(string) == ""` was true before, and `is_live` was defined we would write it to the hash code. Now `is_live` can't exist, so we never write anything if `withStateV.(string) == ""`

Sounds good to me!